src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingServiceConfig.java

```diff
@@ -37,6 +44,60 @@ public interface ServiceConfig {
     String apiKey();
   }
 
+  interface VectorProviderConfig {
+    @JsonProperty
+    boolean enabled();
+
+    @JsonProperty
+    String url();
+
+    @JsonProperty
+    List<String> supportedAuthentication();
+
+    @JsonProperty
+    Map<String, ParameterConfig> parameters();
+
+    @JsonProperty
+    PropertiesConfig properties();
+
+    @JsonProperty
+    Map<String, ModelConfig> models();
+
+    interface ParameterConfig {
+      @JsonProperty
+      String type();
+
+      @JsonProperty
+      boolean required();
+
+      @JsonProperty
+      @WithName("default")
+      Optional<String> default_();
+    }
+
+    interface PropertiesConfig {
+      @JsonProperty
+      Optional<Integer> maxRetries();
+
+      @JsonProperty
+      Optional<Integer> retryDelayMs();
+
+      @JsonProperty
+      Optional<Integer> timeoutMs();
+
+      @JsonProperty
+      Optional<Integer> maxTokens();
```

Comment:
I don't think we have a valid logic to check for token size. Would keep it to maxFieldSize().